pkg/artifactory/repositories.go

```diff
@@ -1271,199 +1267,52 @@ func isSelectRandom(opts ...bool) bool {
+	"nuget":     {RepoLayoutRef: "nuget-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"opkg":      {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"p2":        {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"remote": true, "virtual": true}},
+	"pub":       {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"puppet":    {RepoLayoutRef: "puppet-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"pypi":      {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"sbt":       {RepoLayoutRef: "sbt-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
+	"vagrant":   {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "federated": true}},
+	"vcs":       {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"remote": true}},
+	"rpm":       {RepoLayoutRef: "simple-default", SupportedRepoTypes: map[string]bool{"local": true, "remote": true, "virtual": true, "federated": true}},
 }
 
-//Return the default repo layout of Federated Repository type
-func getDefaultFederatedRepoLayoutRef(repoType string, opts ...bool) string {
-
-	selectRandom := isSelectRandom(opts...)
-	const defaultRepoLayout = "simple-default"
-	repoLayout := defaultRepoLayout
-
-	federatedDefaultRepoLayoutMap := map[string]string{
-		"alpine":    "simple-default",
-		"bower":     "bower-default",
-		"cran":      "simple-default",
-		"cargo":     "simple-default",
-		"chef":      "simple-default",
-		"cocoapods": "simple-default",
-		"composer":  "composer-default",
-		"conan":     "conan-default",
-		"conda":     "simple-default",
-		"debian":    "simple-default",
-		"docker":    "simple-default",
-		"gems":      "simple-default",
-		"generic":   "simple-default",
-		"gitlfs":    "simple-default",
-		"go":        "go-default",
-		"gradle":    "maven-2-default",
-		"helm":      "simple-default",
-		"ivy":       "ivy-default",
-		"maven":     "maven-2-default",
-		"npm":       "npm-default",
-		"nuget":     "nuget-default",
-		"opkg":      "simple-default",
-		"pub":       "simple-default",
-		"puppet":    "puppet-default",
-		"pypi":      "simple-default",
-		"sbt":       "sbt-default",
-		"vagrant":   "simple-default",
-		"rpm":       "simple-default",
-	}
-
-	if selectRandom {
-		repoLayout = selectRandomFromMapOfStrings(federatedDefaultRepoLayoutMap)
+//Return the default repo layout by Repository Type & Package Type
+func getDefaultRepoLayoutRef(repositoryType string, packageType string) string {
```

Comment:
Let's call it `repositoryClass` instead.

Reply:
function `selectRandomFromMapOfStrings` has been removed.